2024-05-27 23:20:19,228 - MainThread - sqlmesh.core.config.connection - INFO - Creating new DuckDB adapter for data files: {'db.db'} (connection.py:294)
2024-05-27 23:20:19,263 - MainThread - sqlmesh.core.renderer - WARNING - Column '"all"' could not be resolved for model '"db"."bps"."compras"', the column may not exist or is ambiguous (renderer.py:506)
2024-05-27 23:20:19,464 - MainThread - sqlmesh.core.config.connection - INFO - Using existing DuckDB adapter due to overlapping data file: db.db (connection.py:290)
2024-05-27 23:20:19,577 - MainThread - sqlmesh.core.snapshot.evaluator - INFO - Listing data objects in schema db.sqlmesh__dtb (evaluator.py:265)
2024-05-27 23:20:19,577 - MainThread - sqlmesh.core.engine_adapter.base - INFO - Executing SQL: SELECT CURRENT_CATALOG() (base.py:1822)
2024-05-27 23:20:19,578 - MainThread - sqlmesh.core.engine_adapter.base - INFO - Executing SQL: SELECT CURRENT_CATALOG() (base.py:1822)
2024-05-27 23:20:19,578 - MainThread - sqlmesh.core.engine_adapter.base - INFO - Executing SQL: SELECT table_name AS name, table_schema AS schema, CASE table_type WHEN 'BASE TABLE' THEN 'table' WHEN 'VIEW' THEN 'view' WHEN 'LOCAL TEMPORARY' THEN 'table' END AS type FROM information_schema.tables WHERE table_schema = 'sqlmesh__dtb' AND table_name IN ('dtb__territorio__3564516099', 'dtb__territorio__3564516099__temp') (base.py:1822)
2024-05-27 23:20:19,599 - MainThread - sqlmesh.core.snapshot.evaluator - INFO - Listing data objects in schema db.sqlmesh__bps (evaluator.py:265)
2024-05-27 23:20:19,599 - MainThread - sqlmesh.core.engine_adapter.base - INFO - Executing SQL: SELECT CURRENT_CATALOG() (base.py:1822)
2024-05-27 23:20:19,599 - MainThread - sqlmesh.core.engine_adapter.base - INFO - Executing SQL: SELECT CURRENT_CATALOG() (base.py:1822)
2024-05-27 23:20:19,600 - MainThread - sqlmesh.core.engine_adapter.base - INFO - Executing SQL: SELECT table_name AS name, table_schema AS schema, CASE table_type WHEN 'BASE TABLE' THEN 'table' WHEN 'VIEW' THEN 'view' WHEN 'LOCAL TEMPORARY' THEN 'table' END AS type FROM information_schema.tables WHERE table_schema = 'sqlmesh__bps' AND table_name IN ('bps__materiais__3533423352', 'bps__unidades__25746938', 'bps__compras__1464341630__temp', 'bps__compras__1464341630', 'bps__materiais__3533423352__temp', 'bps__unidades__25746938__temp') (base.py:1822)
2024-05-27 23:20:19,605 - MainThread - sqlmesh.core.snapshot.evaluator - INFO - Listing data objects in schema db.sqlmesh__siasg (evaluator.py:265)
2024-05-27 23:20:19,605 - MainThread - sqlmesh.core.engine_adapter.base - INFO - Executing SQL: SELECT CURRENT_CATALOG() (base.py:1822)
2024-05-27 23:20:19,605 - MainThread - sqlmesh.core.engine_adapter.base - INFO - Executing SQL: SELECT CURRENT_CATALOG() (base.py:1822)
2024-05-27 23:20:19,606 - MainThread - sqlmesh.core.engine_adapter.base - INFO - Executing SQL: SELECT table_name AS name, table_schema AS schema, CASE table_type WHEN 'BASE TABLE' THEN 'table' WHEN 'VIEW' THEN 'view' WHEN 'LOCAL TEMPORARY' THEN 'table' END AS type FROM information_schema.tables WHERE table_schema = 'sqlmesh__siasg' AND table_name IN ('siasg__materiais__382104304', 'siasg__uasgs__3746688468__temp', 'siasg__unidades__3458370508', 'siasg__uasgs__3746688468', 'siasg__unidades__3458370508__temp', 'siasg__compras__1111392831', 'siasg__compras__1111392831__temp', 'siasg__materiais__382104304__temp') (base.py:1822)
2024-05-27 23:20:19,614 - MainThread - sqlmesh.core.snapshot.evaluator - INFO - Creating schema 'db.sqlmesh__siasg' (evaluator.py:849)
2024-05-27 23:20:19,614 - MainThread - sqlmesh.core.engine_adapter.base - INFO - Executing SQL: CREATE SCHEMA IF NOT EXISTS "db"."sqlmesh__siasg" (base.py:1822)
2024-05-27 23:20:19,614 - MainThread - sqlmesh.core.snapshot.evaluator - INFO - Creating schema 'db.sqlmesh__dtb' (evaluator.py:849)
2024-05-27 23:20:19,614 - MainThread - sqlmesh.core.engine_adapter.base - INFO - Executing SQL: CREATE SCHEMA IF NOT EXISTS "db"."sqlmesh__dtb" (base.py:1822)
2024-05-27 23:20:19,614 - MainThread - sqlmesh.core.snapshot.evaluator - INFO - Creating schema 'db.sqlmesh__bps' (evaluator.py:849)
2024-05-27 23:20:19,614 - MainThread - sqlmesh.core.engine_adapter.base - INFO - Executing SQL: CREATE SCHEMA IF NOT EXISTS "db"."sqlmesh__bps" (base.py:1822)
2024-05-27 23:20:19,619 - MainThread - sqlmesh.core.renderer - WARNING - Column '"all"' could not be resolved for model '"db"."bps"."compras"', the column may not exist or is ambiguous (renderer.py:506)
2024-05-27 23:20:19,621 - MainThread - sqlmesh.core.snapshot.evaluator - INFO - Creating table 'db.sqlmesh__bps.bps__compras__1464341630__temp' (evaluator.py:1155)
2024-05-27 23:20:19,622 - MainThread - sqlmesh.core.engine_adapter.base - INFO - Executing SQL: CREATE TABLE IF NOT EXISTS "db"."sqlmesh__bps"."bps__compras__1464341630__temp" AS SELECT CAST("data_compra" AS DATE) AS "data_compra", "codigo_compra" AS "codigo_compra", "codigo_item_compra" AS "codigo_item_compra", "codigo_item_catalogo" AS "codigo_item_catalogo", MD5(CONCAT("codigo_pdm", ';', "sigla_tipo_unidade_fornecimento", ';', "unidade_medida", ';', "capacidade_unidade_fornecimento")) AS "codigo_unidade_fornecimento", "quantidade" AS "quantidade", CAST("preco_unitario" AS DECIMAL(18, 2)) AS "preco_unitario", "codigo_uasg" AS "codigo_uasg", "cnpj_fornecedor" AS "cnpj_fornecedor", "fornecedor" AS "fornecedor", "marca_fabricante" AS "marca_fabricante" FROM (SELECT DISTINCT "c".*, "m"."codigo_pdm", "m"."descricao_item" FROM (SELECT "idCompra" AS "codigo_compra", "idItemCompra" AS "codigo_item_compra", "codigoItemCatalogo" AS "codigo_item_catalogo", "siglaUnidadeFornecimento" AS "sigla_tipo_unidade_fornecimento", "nomeUnidadeFornecimento" AS "tipo_unidade_fornecimento", "siglaUnidadeMedida" AS "unidade_medida", "capacidadeUnidadeFornecimento" AS "capacidade_unidade_fornecimento", "quantidade" AS "quantidade", "precoUnitario" AS "preco_unitario", "niFornecedor" AS "cnpj_fornecedor", "nomeFornecedor" AS "fornecedor", "marca" AS "marca_fabricante", "codigoUasg" AS "codigo_uasg", "dataCompra" AS "data_compra" FROM "db"."sqlmesh__siasg"."siasg__compras__1111392831" AS "compras" WHERE FALSE LIMIT 0) AS "c" LEFT JOIN "db"."sqlmesh__bps"."bps__materiais__3533423352" AS "m" ON "c"."codigo_item_catalogo" = "m"."codigo_item_catalogo" WHERE FALSE LIMIT 0) AS "_q_0" WHERE FALSE ORDER BY "all" LIMIT 0 (base.py:1822)
2024-05-27 23:20:19,628 - MainThread - sqlmesh.core.context - ERROR - Apply Failure: Traceback (most recent call last):
  File "/Users/fredguth/Code/sus/sqlmesh/bps/.bps-sqlmesh-env/lib/python3.11/site-packages/sqlmesh/utils/concurrency.py", line 227, in sequential_apply_to_dag
    fn(node)
  File "/Users/fredguth/Code/sus/sqlmesh/bps/.bps-sqlmesh-env/lib/python3.11/site-packages/sqlmesh/utils/concurrency.py", line 165, in <lambda>
    lambda s_id: fn(snapshots_by_id[s_id]),
                 ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/fredguth/Code/sus/sqlmesh/bps/.bps-sqlmesh-env/lib/python3.11/site-packages/sqlmesh/core/snapshot/evaluator.py", line 292, in <lambda>
    lambda s: self._create_snapshot(
              ^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/fredguth/Code/sus/sqlmesh/bps/.bps-sqlmesh-env/lib/python3.11/site-packages/sqlmesh/core/snapshot/evaluator.py", line 679, in _create_snapshot
    evaluation_strategy.create(
  File "/Users/fredguth/Code/sus/sqlmesh/bps/.bps-sqlmesh-env/lib/python3.11/site-packages/sqlmesh/core/snapshot/evaluator.py", line 1178, in create
    self.adapter.ctas(
  File "/Users/fredguth/Code/sus/sqlmesh/bps/.bps-sqlmesh-env/lib/python3.11/site-packages/sqlmesh/core/engine_adapter/shared.py", line 260, in internal_wrapper
    return func(*list_args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/fredguth/Code/sus/sqlmesh/bps/.bps-sqlmesh-env/lib/python3.11/site-packages/sqlmesh/core/engine_adapter/base.py", line 445, in ctas
    return self._create_table_from_source_queries(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/fredguth/Code/sus/sqlmesh/bps/.bps-sqlmesh-env/lib/python3.11/site-packages/sqlmesh/core/engine_adapter/base.py", line 634, in _create_table_from_source_queries
    self._create_table(
  File "/Users/fredguth/Code/sus/sqlmesh/bps/.bps-sqlmesh-env/lib/python3.11/site-packages/sqlmesh/core/engine_adapter/base.py", line 670, in _create_table
    self.execute(
  File "/Users/fredguth/Code/sus/sqlmesh/bps/.bps-sqlmesh-env/lib/python3.11/site-packages/sqlmesh/core/engine_adapter/base.py", line 1819, in execute
    self._execute(sql, **kwargs)
  File "/Users/fredguth/Code/sus/sqlmesh/bps/.bps-sqlmesh-env/lib/python3.11/site-packages/sqlmesh/core/engine_adapter/base.py", line 1825, in _execute
    self.cursor.execute(sql, **kwargs)
duckdb.duckdb.BinderException: Binder Error: Referenced column "all" not found in FROM clause!
Candidate bindings: "_q_0.quantidade"
LINE 1: ...IMIT 0) AS "_q_0" WHERE FALSE ORDER BY "all" LIMIT 0
                                                  ^

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/fredguth/Code/sus/sqlmesh/bps/.bps-sqlmesh-env/lib/python3.11/site-packages/sqlmesh/core/context.py", line 1154, in apply
    self._apply(plan, circuit_breaker)
  File "/Users/fredguth/Code/sus/sqlmesh/bps/.bps-sqlmesh-env/lib/python3.11/site-packages/sqlmesh/core/context.py", line 1697, in _apply
    self._scheduler.create_plan_evaluator(self).evaluate(plan, circuit_breaker=circuit_breaker)
  File "/Users/fredguth/Code/sus/sqlmesh/bps/.bps-sqlmesh-env/lib/python3.11/site-packages/sqlmesh/core/plan/evaluator.py", line 109, in evaluate
    self._push(plan, deployability_index_for_creation)
  File "/Users/fredguth/Code/sus/sqlmesh/bps/.bps-sqlmesh-env/lib/python3.11/site-packages/sqlmesh/core/plan/evaluator.py", line 199, in _push
    self.snapshot_evaluator.create(
  File "/Users/fredguth/Code/sus/sqlmesh/bps/.bps-sqlmesh-env/lib/python3.11/site-packages/sqlmesh/core/snapshot/evaluator.py", line 290, in create
    concurrent_apply_to_snapshots(
  File "/Users/fredguth/Code/sus/sqlmesh/bps/.bps-sqlmesh-env/lib/python3.11/site-packages/sqlmesh/utils/concurrency.py", line 163, in concurrent_apply_to_snapshots
    return concurrent_apply_to_dag(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/fredguth/Code/sus/sqlmesh/bps/.bps-sqlmesh-env/lib/python3.11/site-packages/sqlmesh/utils/concurrency.py", line 198, in concurrent_apply_to_dag
    return sequential_apply_to_dag(dag, fn, raise_on_error)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/fredguth/Code/sus/sqlmesh/bps/.bps-sqlmesh-env/lib/python3.11/site-packages/sqlmesh/utils/concurrency.py", line 230, in sequential_apply_to_dag
    raise NodeExecutionFailedError(node) from ex
sqlmesh.utils.concurrency.NodeExecutionFailedError: Execution failed for node SnapshotId<"db"."bps"."compras": 507836180>
 (context.py:1162)
2024-05-27 23:20:19,630 - MainThread - root - INFO - Shutting down the event dispatcher (dispatcher.py:159)
